src/NServiceBus.Core/Persistence/Learning/SagaPersister/SaveAction.cs

```diff
@@ -0,0 +1,42 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Sagas;
+
+    class SaveAction : StorageAction
+    {
+        SagaCorrelationProperty correlationProperty;
+
+        public SaveAction(SagaCorrelationProperty correlationProperty, IContainSagaData sagaData, Dictionary<string, SagaStorageFile> sagaFiles, SagaManifestCollection sagaManifests) : base(sagaData, sagaFiles, sagaManifests)
+        {
+            this.correlationProperty = correlationProperty;
+        }
+
+        public override async Task Execute()
+        {
+            var sagaId = sagaData.Id;
+            var sagaManifest = sagaManifests.GetForEntityType(sagaData.GetType());
+
+            try
+            {
+                var sagaFile = SagaStorageFile.Create(sagaId, sagaManifest);
+                sagaFiles.RegisterSagaFile(sagaFile, sagaId, sagaManifest.SagaEntityType);
+
+                await sagaFile.Write(sagaData)
+                    .ConfigureAwait(false);
+            }
+            catch (Exception ex) when(ex is ConcurrencyException || ex is IOException)
+            {
+                if (correlationProperty == SagaCorrelationProperty.None)
```

Comment:
This should never happen with this persister since each saga must have a single correlation property